README.md

```diff
@@ -0,0 +1,492 @@
+
+The unit test framework used is *pytest* and you can run your test suite using the *Nox* session ``tests`` :
+
+```
+nos -s tests -v
+```
+
+##### Functional tests
+
+Each model project should embed functional tests in the ``ci`` folder, that ensure the training code is purely functional, i.e. without any precision or performance consideration.
+
+The principle of this functional test is to launch a training on a dummy dataset. To be sure to detect possible regression due to changes in the code, and not on the data source side, the dummy dataset must be generated synthetically. 
+
+The ``ci`` folder must contains 
+- a ``generate_synthetic_data.py`` file in charge of the synthetic dataset generation based on the real data format. This script must taking care of not erasing real data already present in the ``data`` folder. We sugges to raise an exception with an explicit message in case of presence of existing data in the targetted path. Future work, should address this issue making the data path configurable.
+- a ``configs`` folder with the ligthning formated configuration file of the test training,
+- a ``run.sh`` script file in charge of running the different training tests.
+
+Optionally it can alos includes :
+- a ``requirements_data.txt`` listing the dependencies related to the synthetic data generation.
+
+The *Nox* session ``train_test`` allows to run the functional tests :
+```
+nox -s train_test -v
+```
+
+By default, the synthetic data are kept in the ``data`` folder, after the tests had run. To clean the ``data`` folder at the end of the tests, use the ``clean_data`` option :
+```
+nox -s train_test -v -- clean_data
+```
+
+### Pull request
+
+Now, your development are done, the last step is to open a *Pull Request* to propose your contribution.
+
+#### Document the model project
+
+In each model project a ``README.md`` file should :
+
+- introduce the scientific challenge adressed,
+- how to get the dataset,
+- briefly describe the model input and output variables and the NN architecture implemented.
+
+#### Finalize the *requirements.txt*
+
+Before creating the *Pull Request* itself, you must generate the final *requirements.txt* file to freeze all the python dependencies of your model project.
+
+In your developemnt environment use :
+```
+pip freeze > requirements.txt
+```
+
+Then commit this last change.
+
+#### Add the model project to the CI/CD workflows
+
+The CI/CD workflows are defined by the Github Actions Yaml files in the ``workflows`` folder. There are 2 different workflow :
+- the *code quality* CI/CD workflow : this workflow (see ``workflows/ai4sim-ci-cd.yaml``) is in charge of running tests and lint for all model project, to detect possible regressions. It is triggered on any changed done on a any branch and on any pull requests.
```

Comment:
for all model projects ... it is triggered by any change done on any branch and on any pull request.